Prevent crashes caused by characters in SIDs which are invalid in paths

EuniverseCat opened the pull request.

EuniverseCat committed:
Prevent crashes caused by characters in SIDs which are invalid in paths

0x0ade commented on Celeste.Mod.mm/Patches/MountainModel.cs:
```diff
@@ -74,7 +74,13 @@ class patch_MountainModel : MountainModel {
+            string path;
+            try {
+                path = Path.Combine("Maps", SaveData.Instance?.LastArea.GetSID() ?? "").Replace('\\', '/');
+            }
+            catch (ArgumentException) {
```
```suggestion
            } catch (ArgumentException) {
```

0x0ade commented on Celeste.Mod.mm/Patches/MountainModel.cs:
```diff
@@ -98,7 +104,13 @@ class patch_MountainModel : MountainModel {
+            string path;
+            try {
+                path = Path.Combine("Maps", SaveData.Instance?.LastArea.GetSID() ?? "").Replace('\\', '/');
+            }
+            catch (ArgumentException) {
```
```suggestion
            } catch (ArgumentException) {
```

0x0ade commented on Celeste.Mod.mm/Patches/MountainModel.cs:
```diff
@@ -112,7 +124,13 @@ class patch_MountainModel : MountainModel {
+                string oldPath;
+                try {
+                    path = Path.Combine("Maps", PreviousSID ?? "").Replace('\\', '/');
+                }
+                catch (ArgumentException) {
```
```suggestion
                } catch (ArgumentException) {
```

0x0ade commented on Celeste.Mod.mm/Patches/MountainModel.cs:
```diff
@@ -112,7 +124,13 @@ class patch_MountainModel : MountainModel {
                         newMountain = meta.Mountain;
                     }
                 }
-                string oldPath = Path.Combine("Maps", PreviousSID ?? "").Replace('\\', '/');
+                string oldPath;
+                try {
+                    path = Path.Combine("Maps", PreviousSID ?? "").Replace('\\', '/');
+                }
+                catch (ArgumentException) {
+                    path = "Maps";
+                }
```
This change seems to cause a build failure. Did you mean to assign to `oldPath` instead of `path` here?

EuniverseCat replied:
you ever make a typo in your first pr to a repo that more than 2 people work on

EuniverseCat committed:
Update Celeste.Mod.mm/Patches/MountainModel.cs

Co-authored-by: 0x0ade <nuda1998@gmail.com>

EuniverseCat committed:
Update Celeste.Mod.mm/Patches/MountainModel.cs

Co-authored-by: 0x0ade <nuda1998@gmail.com>

EuniverseCat committed:
Update Celeste.Mod.mm/Patches/MountainModel.cs

Co-authored-by: 0x0ade <nuda1998@gmail.com>

EuniverseCat committed:
fix build error

today i will make a pr to the everest repo *figures i can't mess up 3 lines of code too badly and doesn't actually compile it myself*

EuniverseCat committed:
how did that even get deleted???? next time i will leave this to max
